src/main/java/org/candlepin/subscriptions/tally/CombiningRollupSnapshotStrategy.java

```diff
@@ -269,7 +267,22 @@ private TallySnapshotNaturalKey calculateRollupKey(
       Map<OffsetDateTime, AccountUsageCalculation> accountCalcs,
       Granularity granularity) {
 
-    List<TallySnapshot> saved = new ArrayList<>();
+    List<TallySnapshot> toSave = new ArrayList<>();
+
+    Map<TallySnapshotNaturalKey, TallySnapshot> affectedSnaps =
+        existingSnapshotLookup.entrySet().stream()
+            .filter(existing -> existing.getValue().getGranularity() == granularity)
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+    // Reset the measurements for all existing finest granularity snapshots so that
+    // they reflect the values of the incoming account calculations.
+    affectedSnaps.values().stream()
+        .forEach(
+            existing ->
+                existing
+                    .getTallyMeasurements()
+                    .keySet()
+                    .forEach(key -> existing.getTallyMeasurements().put(key, 0.0)));
```

Comment:
Why not simply
```java
    affectedSnaps.values().stream()
        .forEach(
            existing ->
                existing
                    .getTallyMeasurements()
                        .clear()
```
?

Reply:
I did this because I thought that Marketplace would need the measurements to exist in the snapshots to identify that the value was changed to 0 instead of not being present. Happy to change this if this isn't the case.

Reply:
Actually, so long as the snapshot record itself is still around (w/ consistent UUID), shouldn't matter.